### PR TITLE
Map Tools - Fix tools showing after item is dropped out of inventory

### LINKED ISSUE #9748: Map Tools - Maptools remains when item is dropped out of inventory. (WIP)
**Mods (complete and add to the following information):**
- **Arma 3:** `2.14` current stable
- **CBA:** `3.16.1` current stable 
- **ACE3:** `v3.16.3` current stable

^ will do later

**Description:**

During the process of exploring and playing around with the plotting board i also had the map tool open. to get the map markers on the plotting board to move, (unsuccessfully), i dropped the maptools while the maptools ahve been on the map. (together with the PB)

After dropping the MapTools, the MapTool Image remained on the Map but i was not able to interact with it ( Move, Turn, ace map interaction, ...)

Once I picked up the map tools again, i was able to move the MT around the map again and interact with it as ususal.

**Steps to reproduce:** // TBA
_Follow [https://ace3.acemod.org/img/wiki/user/issue_flowchart.webp](this flowchart)!_

1. _Go to ..._
2. _Click ..._
3. _See ..._

**Expected behavior:**
When i drop the Map Tools out of my inventory while i have 0 remaining maptools, i expect the maptools to disappear from the map, since they are gone.

**Where did the issue occur?**
- Dedicated Server

**Log Files:**
Will add Log File once i was able to reproduce with CBA, ACE only.

**Additional context:**
Issue found while doing other stuff, didnt had the time reproduce with reduced modlist yet.

**Screenshots:**
tba.

### PR DESCRIPTION
**Pull request by PabstMirror**
Close #9748

### CODE REVIEW

**Review by LinkIsGrim — APPROVED** [review]

File: addons/maptools/functions/fnc_updateMapToolMarkers.sqf [hunk 1]
```diff
@@ -25,15 +25,8 @@ if (GVAR(plottingBoard_Shown) == 0) then {
...
-
-if (((GVAR(mapTool_Shown) == 0) && {GVAR(plottingBoard_Shown) == 0}) || {
-    private _uniqueItems = ACE_player call EFUNC(common,uniqueItems);
-
-    !(("ACE_MapTools" in _uniqueItems) || {"ACE_PlottingBoard" in _uniqueItems})
-}) exitWith {};
-
-if (GVAR(plottingBoard_Shown) > 0) then {
+} else {
+    if (!("ACE_PlottingBoard" in (ACE_player call EFUNC(common,uniqueItems)))) exitWith {};
```
Comment (johnb432) [inline addons/maptools/functions/fnc_updateMapToolMarkers.sqf:29]:
```suggestion
    if !("ACE_PlottingBoard" call EFUNC(common,hasItem)) exitWith {};
```

File: addons/maptools/functions/fnc_updateMapToolMarkers.sqf [hunk 2]
```diff
@@ -97,7 +90,7 @@ if (GVAR(plottingBoard_Shown) > 0) then {
     } forEach GVAR(plottingBoard_markers);
 };
 
-if (GVAR(mapTool_Shown) > 0) then {
+if ((GVAR(mapTool_Shown) > 0) && {"ACE_MapTools" in (ACE_player call EFUNC(common,uniqueItems))}) then {
```
Comment (johnb432) [inline addons/maptools/functions/fnc_updateMapToolMarkers.sqf:93]:
```suggestion
if ((GVAR(mapTool_Shown) > 0) && {"ACE_MapTools" call EFUNC(common,hasItem)}) then {
```

File: addons/maptools/functions/fnc_updateMapToolMarkers.sqf (hunk 1 above)
Comment (PabstMirror) [reply addons/maptools/functions/fnc_updateMapToolMarkers.sqf:29]: https://github.com/acemod/ACE3/blob/master/addons/common/functions/fnc_hasItem.sqf
takes 2 args

Comment (johnb432) [reply addons/maptools/functions/fnc_updateMapToolMarkers.sqf:29]: I can't believe I looked up the function before posting _and still got it wrong_. Updated suggestion.

**Pull request merged**